actionview/lib/action_view/template.rb

```diff
@@ -297,20 +307,6 @@ def #{method_name}(local_assigns, output_buffer)
         ObjectSpace.define_finalizer(self, Finalizer[method_name, mod])
       end
 
-      def handle_render_error(view, e) #:nodoc:
-        if e.is_a?(Template::Error)
-          e.sub_template_of(self)
-          raise e
-        else
-          template = self
-          unless template.source
-            template = refresh(view)
-            template.encode!
-          end
-          raise Template::Error.new(template, e)
-        end
-      end
-
```

Comment:
Let's move this back to the `handle_render_error` method. Then template subclasses can continue to override it.